src/js/modules/articles/selectors.js

```diff
@@ -1,26 +1,127 @@
+import React from 'react';
 import { createSelector } from "reselect";
-import { getSectionAndSubsectionSlugs } from "../sections/selectors";
 
-const getArticles = state => state.articles.articles;
+import { getUsers, getContributorFromSlug } from "../users/selectors";
+import {
+  getSections,
+  getSectionFromProps,
+  getSectionSlugFromId,
+  getSlugsInSectionTree,
+} from "../sections/selectors";
+
+export const getArticles = state => state.articles.articles;
+const getAuthorships = state => state.articles.authorships;
+const getArticlesResponse = state => state.articles.response;
 const getRequestedArticleSlug = (state, props) => props.match.params.article_slug;
-const getSectionByProps = (state, props) => props.section;
 
-export const getArticleBySlug = createSelector(
-  [ getSectionByProps, getArticles, getRequestedArticleSlug ],
-  (section, articles, articleSlug) => {
-    const requestedArticle = articles[ articleSlug ];
-    // the article may match the slug, but does it match the requested section?
+/**
+ * The selector returns an articles object with an additional property for each
+ *   article: its byline in JSX.
+ */
+const getArticlesWithContributors = createSelector(
+  [ getArticles, getUsers, getAuthorships ],
+  (articles, users, authorships) => {
+    let articlesWithContributors = articles;
```

Comment:
I looped through authorships in the interest of efficiency. If i looped through users or articles, I would be going through all authorships several times.